src/helper/service-utils.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+namespace EE\Service\Utils;
+
+use EE;
+use Symfony\Component\Filesystem\Filesystem;
+
+/**
+ * Boots up the container if it is stopped or not running.
+ * @throws EE\ExitException
+ */
+function nginx_proxy_check() {
+
+	$proxy_type = EE_PROXY_TYPE;
+	if ( 'running' !== EE::docker()::container_status( $proxy_type ) ) {
+		/**
+		 * Checking ports.
+		 */
+		$port_80_status  = get_curl_info( 'localhost', 80, true );
+		$port_443_status = get_curl_info( 'localhost', 443, true );
+
+		// if any/both the port/s is/are occupied.
+		if ( ! ( $port_80_status && $port_443_status ) ) {
+			EE::error( 'Cannot create/start proxy container. Please make sure port 80 and 443 are free.' );
+		} else {
+
+			$fs = new Filesystem();
+
+			if ( ! $fs->exists( EE_CONF_ROOT . '/docker-compose.yml' ) ) {
+				generate_global_docker_compose_yml( $fs );
+			}
+
+			$EE_CONF_ROOT = EE_CONF_ROOT;
+			if ( ! EE::docker()::docker_network_exists( GLOBAL_BACKEND_NETWORK ) ) {
```

Comment:
Why did you use two if conditions one after another instead of just using && ?
```php
if (
 ! EE::docker()::docker_network_exists( GLOBAL_BACKEND_NETWORK ) &&
 ! EE::docker()::create_network( GLOBAL_BACKEND_NETWORK ) 
) { 
```